docs/DeveloperGuide.md

```diff
@@ -6,33 +6,360 @@
 
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### User Interface (UI) - Class Implementation
+![img.png](img.png) <br>
+**Figure 1: UML Diagram of User Interface (UI) Class**
+1. The **`UI class`** named `Ui` belongs to the package functionalities.ui. The class contains several static and non-static methods that display messages to the user such as **`showUserMessage()`**, **`showErrorMessage()`**, and **`showWelcomeMessage()`**. The class is also used to read user input, format and print appointment lists, and add or remove appointments from the **`sniff appointment manager`**. The **`Ui`** class has a **`showLine()`** method that displays a divider line to the user.
+2. The **`UI class`** has a private static final String DOT_THEN_SPACE field that is used as a constant string value to format user messages. It has a **`readUserCommand()`** method that returns a string value that is entered by the user through the Command Line Interface (CLI).
+3. The class imports the **`SniffException`** class from the exception package and utilizes it in the **`showErrorMessage()`** method to display an error message to the user.
 
+### Command - Class Implementation
+![img_16.png](Updated%20Command%20UML.png) <br>
+**Figure 2: UML Diagram of Command Class** <br>
+1. The **`Command class`** named `Command` belongs to the package functionalities.commands. The class is an abstract class that provides a basic template for implementing commands in the command-line interface. This class is designed to be extended by subclasses that implement specific commands, such as **`ListCommand`** and **`ConsultationCommand`**.
+2. The **`Command class`** has a boolean isExit field that can be set to true to exit the entire programme, or remain as false to continue running the programme. It has a **`isExit()`** method that sets that field to false initially. It also has a **`executeCommand()`** method that takes an instance of SniffTasks as a parameter and throws a SniffException if an error occurs during execution. Subclasses override this method in order to run other commands.
+
+![img_12.png](img_12.png) <br>
+**Figure 3: Sequence Diagram showing the logical implementation of executeCommand() for the Consultation Command**
+
+![img_14.png](img_14.png) <br>
+**Figure 4: Sequence Diagram showing the logical implementation of executeCommand() for the Vaccination Command**
```

Comment:
![image](https://user-images.githubusercontent.com/88373092/228760005-02d1714f-1d3e-4595-8a66-18a606451731.png)

Is your notation for a method call correct? Perhaps you could consider adding parenthesis to the end of each method call .